libs/core/include/core/synchronisation/protected.hpp

```diff
@@ -28,13 +28,13 @@ namespace fetch {
 template <typename T, typename M = Mutex>
 class Protected
 {
-private:
+protected:
```

Comment:
I'd prefer if we stuck with the original design, which uses encapsulation instead of inheritance.